### PR TITLE
Invoke REPL from code

### PR DESCRIPTION
**Pull request by som-snytt**
A FAQ is how to invoke REPL from code.

Document the common entry points.

### CODE REVIEW

**Comment by som-snytt** [comment]
This PR experienced an intermediate failure state, I don't know if it's meaningful:

```
[tut] compiling: ./pt-br/cheatsheets/index.md
[tut] compiling: ./pt-br/tutorials/tour/_posts/2017-02-13-xml-processing.md
Exception in thread "main" java.lang.NullPointerException
	at scala.reflect.internal.SymbolTable.exitingPhase(SymbolTable.scala:256)
	at scala.tools.nsc.interpreter.IMain$Request.x$20$lzycompute(IMain.scala:896)
	at scala.tools.nsc.interpreter.IMain$Request.x$20(IMain.scala:895)
	at scala.tools.nsc.interpreter.IMain$Request.headerPreamble$lzycompute(IMain.scala:895)
at scala.tools.nsc.interpreter.IMain$Request.headerPreamble(IMain.scala:895)
```

File: overviews/repl/embedding.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        import Results._
+        val res = intp.interpret(code) match {
+          case Success => "OK!"
+          case _       => "Sorry, try again."
+        }
+        println(res)
+      }
+      interpret("""println("hello, world")""")
+      interpret("""println(""")
+      interpret("""val who = "world" ; println("hello, $who")""")
```
Comment (lrytz) [inline overviews/repl/embedding.md]: missing interpolator?

Comment (som-snytt) [inline overviews/repl/embedding.md]: Yes, that was to show the warning output. It should probably explain expected output.

**Comment by som-snytt** [comment]
The script println is missing.
```
hello, world
OK!
Sorry, try again.
<console>:13: warning: possible missing interpolator: detected interpolated identifier `$who`
        println("hello, $who")
                ^
error: No warnings can be incurred under -Xfatal-warnings.
Sorry, try again.
Welcome to Scala 2.12.1 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_111).
Type in expressions for evaluation. Or try :help.

scala> :quit
Welcome to Scala 2.12.1 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_111).
Type in expressions for evaluation. Or try :help.

scala> res0: String = hello, world

scala> :quit
Output is Welcome to Scala 2.12.1 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_111).
Type in expressions for evaluation. Or try :help.

scala> hello, world
res0: Int = 42

scala> :quit

```

File: overviews/repl/embedding.md (hunk 1 above)
Comment (lrytz) [inline overviews/repl/embedding.md]: sorry, thanks :)

**Comment by SethTisue** [comment]
@som-snytt is this ready for merge?

**Comment by SethTisue** [comment]
(feel free to use your own merge powers if so)

**Comment by som-snytt** [comment]
Let our powers merge!

The doc will change soon anyway with Friendly Repl. FREPL. Or EPFL FREPL.

If anyone reads this and complains, I will make amends by making emends.

**Comment by SethTisue** [comment]
🎵 Just want to drink my REPL wine, down in the Lightning Bar 🎵

**Pull request merged**